dev/docker/ballista-scheduler.Dockerfile

```diff
@@ -24,7 +24,16 @@ ENV RUST_LOG=info
 ENV RUST_BACKTRACE=full
 ENV DEBIAN_FRONTEND=noninteractive
 
-RUN apt-get update && apt-get install -y nginx netcat
+RUN apt-get update &&\
+    apt-get install -y nginx netcat nodejs npm
+    
+COPY ballista/ui /root/ballista/ui
+
+RUN cd /root/ballista/ui/scheduler &&\
```

Comment:
We deliberately got away from building inside the deployment container. If you run https://github.com/apache/arrow-ballista/blob/master/dev/build-ballista-docker.sh

It should run the build https://github.com/apache/arrow-ballista/blob/231d1e3c9936d44c435d1d2fef54178c77e137d7/dev/build-ballista-docker.sh#L22

who's entrypoint calls yarn build https://github.com/apache/arrow-ballista/blob/231d1e3c9936d44c435d1d2fef54178c77e137d7/dev/docker/ballista-builder.Dockerfile#L29

Reply:
@avantgardnerio Really thanks for explaining! This works. So I guess the real issue is really the docs which I can fix. `. dev/build-ballista-docker.sh` is what people should run.

Reply:
The ARM docker build is removed in https://github.com/apache/arrow-ballista/pull/413

Reply:
@andygrove Thanks for letting me know! So docker build doesn't really work by running any script if one is on ARM and for AMD people `. dev/build-ballista-docker.sh` should be used, right?

Reply:
I'd like to not have any `.sh` files at all. The build should not be complicated:

```
git clone
cargo build
npm build
docker build
```

Every time I see shell scripts, then I see tech debt go into them instead of just making the builds as easy as above. So the only issue here is just that `npm build` didn't go into all the places in the docs where it should, which is my main complaint against docs as well. The CI scripts are the canonical place to look to see how to run a build, and they are under test.

Reply:
Apologies the previous two comments were not productive. I think the way forward is one that @andygrove suggested: generate the docs from the scripts somehow. I'll have a think on that for a bit. Happy to hear about ways to do that if anyone has any advice.

Reply:
> The CI scripts are the canonical place to look to see how to run a build, and they are under test.

For expert developers such as yourself, I can see how this makes sense, but I want to make sure that we are inclusive for all levels and backgrounds. Some developers would feel intimidated if they were expected to learn how CI works just to be able to build the project locally and would prefer to have documentation and scripts, especially with a project this complex, with multiple languages & technologies (Python, Rust, React, Docker). I know I have given up on projects if I couldn't figure out how to build them locally without going through new learning curves.

I'll file an issue for Windows support. It would be great to support that.